libminifi/src/core/Processor.cpp

```diff
@@ -277,6 +286,28 @@ void Processor::onTrigger(const std::shared_ptr<ProcessContext> &context, const
   }
 }
 
+void Processor::callOnTrigger(const std::shared_ptr<ProcessContext>& context, const std::shared_ptr<ProcessSession>& session) {
+  uint64_t num_incoming_flow_files_queued_before = numIncomingFlowFilesQueued();
+  onTrigger(context, session);  // virtual method overridden by subclasses
```

Comment:
The lock is unlocked inbetween the calls. Could there be no issue with subsequent calls to this function on multiple threads? Eg. can this case not happen?
1. Count flowfiles before
1. Call onTrigger
1. Add files to incoming connections on a different thread
1. Count flowfiles after

Reply:
https://github.com/apache/nifi-minifi-cpp/pull/1013#discussion_r580306578

That's what I say, too. :)